sources/packages/backend/apps/api/src/route-controllers/application/application.controller.service.ts

```diff
@@ -775,4 +783,120 @@ export class ApplicationControllerService {
       throw error;
     }
   }
+
+  /**
+   * Validate the application submission and get the validated result.
+   * @param applicationId application ID.
+   * @param studentId student used for authorization.
+   * @param payload payload to create the application.
+   * @throws UnprocessableEntityException if the program year is not active.
+   * @throws BadRequestException if the application data is invalid, failing the druRun submission.
```

Comment:
As we are listing the potential exceptions here, what about NotFoundException?

Reply:
I was planning to list them all, but I am not sure if we should do it as a pattern, to be honest.
I believe that I will remove them all. Please let me know if there is a concern about keeping or removing it.

Reply:
Removed them for now.